### PR TITLE
RDCC-2963: PACT and Functional test to complete for of JRD POST API

### PR DESCRIPTION
**Pull request by ishratt**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-2963

### Change description ###
Added functional tests and pact tests for JRD fetch users post api

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by shreedhar-hmcts — APPROVED** [review]

**Review by shreedhar-hmcts — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/JudicialUsersFunctionalTest.java [hunk 1]
```diff
@@ -84,4 +130,13 @@ private UserRequest getUserRequest() {
         return new UserRequest(userIds);
     }
 
+    private UserRequest getUserRequest() {
+        List<String> userIds = new ArrayList<>();
```
Comment (viswajit-kuna) [inline src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/JudicialUsersFunctionalTest.java:134]: You may use the below:
```
var userIds = new ArrayList<>();
```

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/JudicialUsersFunctionalTest.java [hunk 2]
```diff
@@ -1,60 +1,110 @@
...
+    }
 
     @Test
     @ToggleEnable(mapKey = FETCH_USERS, withFeature = true)
-    public void shouldReturnDataNotFound() {
-        Response fetchResponse = judicialApiClient.getMultipleAuthHeadersInternal(ROLE_JRD_SYSTEM_USER)
-                .body(getUserRequest())
-                .post(PATH)
-                .andReturn();
+    public void shouldReturn200() {
```
Comment (viswajit-kuna) [inline src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/JudicialUsersFunctionalTest.java:75]: It is a good practice to have a descriptive name for the test method. Alternatively, you could use the `@DisplayName` to explain the scenario being tested in the test case. `shouldReturn200` only indicates the outcome but it doesn't tell which scenario is being tested.

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/client/JudicialApiClient.java [hunk 3]
```diff
@@ -92,5 +99,24 @@ public RequestSpecification getMultipleAuthHeaders(String userToken) {
                 .header(AUTHORIZATION_HEADER, "Bearer " + userToken);
     }
 
+    public Object fetchUserProfiles(UserRequest userRequest, int pageSize, int pageNumber, HttpStatus expectedStatus,
+                                    String role) {
+        Response fetchResponse = getMultipleAuthHeadersInternal(role)
+                .body(userRequest).log().body(true)
+                .post("/refdata/judicial/users/fetch?page_size=" + pageSize + "&page_number=" + pageNumber)
```
Comment (viswajit-kuna) [inline src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/client/JudicialApiClient.java:106]: You may declare the URI - `/refdata/judicial/users/fetch?page_size=%s&page_number=%s` as a constant and use `String.format` to replace these placeholders with necessary values.

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/JudicialUsersFunctionalTest.java (hunk 1 above)
Comment (ishratt) [reply src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/JudicialUsersFunctionalTest.java:134]: addressed here dc7b300

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/JudicialUsersFunctionalTest.java (hunk 2 above)
Comment (ishratt) [reply src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/JudicialUsersFunctionalTest.java]: addressed here dc7b300

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/client/JudicialApiClient.java (hunk 3 above)
Comment (ishratt) [reply src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/client/JudicialApiClient.java]: addressed here dc7b300

**Review by shreedhar-hmcts — APPROVED** [review]

**Review by viswajit-kuna — APPROVED** [review]

**Pull request merged**